lib/adbc_connection.ex

```diff
@@ -392,22 +392,35 @@ defmodule Adbc.Connection do
   defp normalize_rows(-1), do: nil
   defp normalize_rows(rows) when is_integer(rows) and rows >= 0, do: rows
 
-  defp stream_results(reference, num_rows), do: stream_results(reference, %{}, num_rows)
+  defp stream_results(reference, num_rows), do: stream_results(reference, [], num_rows)
 
   defp stream_results(reference, acc, num_rows) do
     case Adbc.Nif.adbc_arrow_array_stream_next(reference) do
       {:ok, results, _done} ->
-        acc = Map.merge(acc, Map.new(results), fn _k, v1, v2 -> v1 ++ v2 end)
-        stream_results(reference, acc, num_rows)
+        stream_results(reference, acc ++ results, num_rows)
 
       :end_of_series ->
-        {:ok, %Adbc.Result{data: acc, num_rows: num_rows}}
+        {:ok, %Adbc.Result{data: merge_columns(acc), num_rows: num_rows}}
 
       {:error, reason} ->
         {:error, error_to_exception(reason)}
     end
   end
 
+  defp merge_columns(columns) do
+    Enum.reduce(columns, [], fn column, merged_columns ->
```

Comment:
The columns should always be in the same order, right? Maybe we can use zip_with instead?

Reply:
I also wonder if maybe call `stream_results` passing empty columns as a third argument. Then `adbc_arrow_array_stream_next` returns only a list of lists, which we can concatenate, and put it back into the Adbc.Column only at the end. WDYT? This would be the most efficient way.

Reply:
Alternatively, we can return the empty columns/metadata with `:end_of_series`.

So overall, it would be something like this: on each result set, we get a list of lists. We can either zip them together or collect them to flatten at the end. Then at the end we receive empty column structs from the metadata and put the actual data inside (by using zip_with)?

Reply:
But we probably still need to "traverse" the result if there're any nested list or structs and concatenate these ones...? Maybe we can send paged/chunked results to users by default (the same way as arrow/adbc returns the results to us) and only concatenate the results if user explicitly requested?